packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -0,0 +1,342 @@
+import { updateAdditionalDepositData } from '../util/deposits/helpers'
+
+const PAGE_SIZE = 100
+
+export type Deposit = Transaction
+
+export type Withdrawal =
+  | FetchWithdrawalsFromSubgraphResult
+  | WithdrawalInitiated
+  | EthWithdrawal
+
+type DepositOrWithdrawal = Deposit | Withdrawal
+
+function getStandardizedTimestampByTx(tx: DepositOrWithdrawal) {
+  if (isDeposit(tx)) {
+    return tx.timestampCreated ?? 0
+  }
+
+  if (isWithdrawalFromSubgraph(tx)) {
+    return getStandardizedTimestamp(tx.l2BlockTimestamp)
+  }
+
+  return getStandardizedTimestamp(tx.timestamp ?? '0')
+}
+
+function sortByTimestampDescending(
+  a: DepositOrWithdrawal,
+  b: DepositOrWithdrawal
+) {
+  return getStandardizedTimestampByTx(a) > getStandardizedTimestampByTx(b)
+    ? -1
+    : 1
+}
+
+const multiChainFetchList: { parentChain: ChainId; chain: ChainId }[] = [
+  {
+    parentChain: ChainId.Ethereum,
+    chain: ChainId.ArbitrumOne
+  },
+  // {
+  //   parentChain: ChainId.Mainnet,
+  //   chain: ChainId.ArbitrumNova
+  // },
+  // Testnet
+  {
+    parentChain: ChainId.Goerli,
+    chain: ChainId.ArbitrumGoerli
+  },
+  {
+    parentChain: ChainId.Sepolia,
+    chain: ChainId.ArbitrumSepolia
+  }
+  // Orbit
+  // {
+  //   parentChain: ChainId.ArbitrumGoerli,
+  //   chain: ChainId.XaiTestnet
+  // }
+  // {
+  //   parentChain: ChainId.ArbitrumSepolia,
+  //   chain: ChainId.StylusTestnet
+  // }
+]
+
+function isWithdrawalFromSubgraph(
+  tx: Withdrawal
+): tx is FetchWithdrawalsFromSubgraphResult {
+  return tx.source === 'subgraph'
+}
+
+function isDeposit(tx: DepositOrWithdrawal): tx is Deposit {
+  return tx.direction === 'deposit'
+}
+
+async function transformTransaction(
+  tx: DepositOrWithdrawal
+): Promise<MergedTransaction | undefined> {
+  const parentChainProvider = getProvider(tx.parentChainId)
+  const childChainProvider = getProvider(tx.childChainId)
+
+  if (isDeposit(tx)) {
+    return transformDeposit(
+      await updateAdditionalDepositData({
+        depositTx: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: childChainProvider
+      })
+    )
+  }
+
+  let withdrawal: L2ToL1EventResultPlus | undefined
+
+  if (isWithdrawalFromSubgraph(tx)) {
+    withdrawal = await mapWithdrawalToL2ToL1EventResult({
+      withdrawal: tx,
+      l1Provider: parentChainProvider,
+      l2Provider: childChainProvider
+    })
+  } else {
+    if (isTokenWithdrawal(tx)) {
+      withdrawal = await mapTokenWithdrawalFromEventLogsToL2ToL1EventResult({
+        result: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: childChainProvider
+      })
+    } else {
+      withdrawal = await mapETHWithdrawalToL2ToL1EventResult({
+        event: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: childChainProvider
+      })
+    }
+  }
+
+  if (withdrawal) {
+    return transformWithdrawal(withdrawal)
+  }
```

Comment:
hm what's the default case here? why can it be undefined? tbh I'd rather we throw if this is unexpected